src/interactivity/base/PseudoConsoleWindowAccessibilityProvider.hpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "precomp.h"
+
+namespace Microsoft::Console::Interactivity
+{
+    class PseudoConsoleWindowAccessibilityProvider final :
+        public WRL::RuntimeClass<WRL::RuntimeClassFlags<WRL::ClassicCom | WRL::InhibitFtmBase>, IRawElementProviderSimple>
+    {
+    public:
+        PseudoConsoleWindowAccessibilityProvider() = default;
+        ~PseudoConsoleWindowAccessibilityProvider() = default;
+        HRESULT RuntimeClassInitialize(HWND pseudoConsoleHwnd) noexcept;
+
+        PseudoConsoleWindowAccessibilityProvider(const PseudoConsoleWindowAccessibilityProvider&) = delete;
+        PseudoConsoleWindowAccessibilityProvider(PseudoConsoleWindowAccessibilityProvider&&) = delete;
+        PseudoConsoleWindowAccessibilityProvider& operator=(const PseudoConsoleWindowAccessibilityProvider&) = delete;
+        PseudoConsoleWindowAccessibilityProvider& operator=(PseudoConsoleWindowAccessibilityProvider&&) = delete;
+
+        // IRawElementProviderSimple methods
+        IFACEMETHODIMP get_ProviderOptions(_Out_ ProviderOptions* pOptions) override;
+        IFACEMETHODIMP GetPatternProvider(_In_ PATTERNID iid,
+                                          _COM_Outptr_result_maybenull_ IUnknown** ppInterface) override;
+        IFACEMETHODIMP GetPropertyValue(_In_ PROPERTYID idProp,
+                                        _Out_ VARIANT* pVariant) override;
+        IFACEMETHODIMP get_HostRawElementProvider(_COM_Outptr_result_maybenull_ IRawElementProviderSimple** ppProvider) override;
+
+    private:
+        HWND _pseudoConsoleHwnd;
+
+        const OLECHAR* AutomationPropertyName = L"Internal Console Management Window";

```

Comment:
one last bit: this may be better handled as a `static constexpr auto` in the cpp file

Reply:
(otherwise: you are storing a member in every instance that points to this string, and... i mean, there's only one instance but why bother?)